### PR TITLE
Ensure we consistently broadcast the result of simd dot product

### LINKED ISSUE #99391: Net8 - Vector2.Normalize and old(?) CPU - wrong result
### Description

Vector2.Normalize - wrong result and i think this is Cpu problem (.Net use some unsuported intrinsics?)

### Reproduction Steps

```
Console.WriteLine(Vector2.Normalize(new(0,2)));
```

### Expected behavior

`<0 1>`

### Actual behavior

`<0  ∞>`

### Regression?

In 7 and 8.0.preview.1 - works
8.0.preview.2, rc.1, 8.0.201 ... - issue

### Known Workarounds

_No response_

### Configuration

I reproduced this on intel celeron e3300 and intel p6100
(Win10 and Win7) x64

### Other information

Edit:
copy from 
https://github.com/microsoft/referencesource/blob/master/System.Numerics/System/Numerics/Vector2.cs#L191

```
        public static Vector2 Normalize2(Vector2 value)
        {
            if (Vector.IsHardwareAccelerated)
            {
//-----------------------------------------------------------------------------
		Console.WriteLine("HardwareAccelerated");
//-----------------------------------------------------------------------------
                float length = value.Length();
                return value / length;
            }
            else
            {
                float ls = value.X * value.X + value.Y * value.Y;
                float invNorm = 1.0f / (float)Math.Sqrt((double)ls);

                return new Vector2(
                    value.X * invNorm,
                    value.Y * invNorm);
            }
        }
```
code
```
var v1 = new Vector2(0,2);
Console.WriteLine(Vector2.Normalize(v1));
Console.WriteLine(Normalize2(v1));
```
shows
```
<0  ∞>
HardwareAccelerated
<0  1>
```

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves #99391

### CODE REVIEW

File: src/coreclr/jit/morph.cpp [hunk 1]
```diff
@@ -9916,16 +9916,6 @@ GenTree* Compiler::fgOptimizeHWIntrinsic(GenTreeHWIntrinsic* node)
...
 
-#if defined(TARGET_XARCH)
-            if ((simdSize == 8) && !compOpportunisticallyDependsOn(InstructionSet_SSE41))
-            {
-                // When SSE4.1 isn't supported then Vector2 only needs a single horizontal add
-                // which means the result isn't broadcast across the entire vector and we can't
-                // optimize
-                break;
-            }
-#endif // TARGET_XARCH
```
Comment (tannergooding) [inline src/coreclr/jit/morph.cpp:9927]: This code was incorrect, as we expect that broadcast should always be occurring.

The lowering logic was doing the right thing already for SSE2 and SSE41+ machines, but not for SSE3 or SSSE3 machines where `HorizontalAdd` exists

**Comment by tannergooding** [comment]
This is going to need backport to .NET 8 since its a customer reported codegen issue.

**Comment by tannergooding** [comment]
CC. @dotnet/jit-contrib, this is ready for review

JitStress failures are unrelated and fixed by https://github.com/dotnet/runtime/pull/105889, I've updated the branch and restarted CI so it can pass cleanly now that the fix has been merged (force pushed so this remains trivial to backport).

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**